src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1288,37 +1289,8 @@ public void SendFile(string? fileName, ReadOnlySpan<byte> preBuffer, ReadOnlySpa
         // Sends data to a specific end point, starting at the indicated location in the buffer.
         public int SendTo(byte[] buffer, int offset, int size, SocketFlags socketFlags, EndPoint remoteEP)
         {
-            ThrowIfDisposed();
-
             ValidateBufferArguments(buffer, offset, size);
-            ArgumentNullException.ThrowIfNull(remoteEP);
-
-            ValidateBlockingMode();
-            if (NetEventSource.Log.IsEnabled()) NetEventSource.Info(this, $"SRC:{LocalEndPoint} size:{size} remoteEP:{remoteEP}");
-
-            Internals.SocketAddress socketAddress = Serialize(ref remoteEP);
-
-            int bytesTransferred;
-            SocketError errorCode = SocketPal.SendTo(_handle, buffer, offset, size, socketFlags, socketAddress.Buffer, socketAddress.Size, out bytesTransferred);
-
-            // Throw an appropriate SocketException if the native call fails.
-            if (errorCode != SocketError.Success)
-            {
-                UpdateSendSocketErrorForDisposed(ref errorCode);
-
-                UpdateStatusAfterSocketErrorAndThrowException(errorCode);
-            }
-            else if (SocketsTelemetry.Log.IsEnabled())
-            {
-                SocketsTelemetry.Log.BytesSent(bytesTransferred);
-                if (SocketType == SocketType.Dgram) SocketsTelemetry.Log.DatagramSent();
-            }
-
-            // Save a copy of the EndPoint so we can use it for Create().
-            _rightEndPoint ??= remoteEP;
-
-            if (NetEventSource.Log.IsEnabled()) NetEventSource.DumpBuffer(this, buffer, offset, size);
-            return bytesTransferred;
+            return SendTo(new ReadOnlySpan<byte>(buffer, offset, size), socketFlags, remoteEP);
```

Comment:
```suggestion
            return SendTo(buffer.AsSpan(offset, size), socketFlags, remoteEP);
```
(though a matter of taste)